scriptmodules/helpers.sh

```diff
@@ -1189,13 +1187,13 @@ function addPort() {
 
     mkUserDir "$romdir/ports"
 
-    if [[ -t 0 ]]; then
+    if [[ -z "$script" ]]; then
         cat >"$file" << _EOF_
 #!/bin/bash
 "$rootdir/supplementary/runcommand/runcommand.sh" 0 _PORT_ "$port" "$game"
 _EOF_
     else
-        cat >"$file"
+        cat >"$file" <<< "$script"
```

Comment:
Perhaps just using cp ?

Reply:
ah - sorry I misunderstood - you are loading the custom scripts into a var and using that as a parameter.  ok.

Reply:
not 100% sure I like this workaround - it might be better for the scriptmodule itself to just copy the custom script in place. However I will keep for now, but may change later.

Reply:
Yes, indeed yesterday I also kept thinking about a very similar idea.
Like patches, these custom launch scripts should be also kept in the `$md_data` directory of the module needing it and the filename passed to `addPort` instead of the content, to be copied accordingly. I agree is better like that.
This would be consistent with how `applyPatch` works too. I will give it a thought and propose a PR. Thanks for the feedback !

Reply:
In the case of the one example we have, it requires variables from the script though, so a separate file would not work like that unless we replaced tokens in it. I was thinking more that the module itself just creates the file - keep it outside of addPort. Eg we have a parameter for addPort like "noscript" which makes it not create a launch script so we can then just do it from the module. But the current way is ok I guess.

Reply:
or we just overwrite the script after calling addPort with the custom one